app/main.py

```diff
@@ -1,2 +1,118 @@
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self) -> int:
+        return round(self.capacity * RESIZE_THRESHOLD)
+
+    def resize(self) -> None:
+        old_hash_table = self.hash_table
+
+        self.__init__(self.capacity * CAPACITY_MULTIPLIER)
+
+        for node in old_hash_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            if self.size + 1 >= self.current_max_size:
+                self.resize()
+                return self.__setitem__(key, value)
+            self.size += 1
+
+        self.hash_table[index] = Node(key, value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find key: {key}")
+
+        return self.hash_table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find key: {key}")
+
+        self.hash_table[index] = None
+        self.size -= 1
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __str__(self) -> str:
+        return str(self.hash_table)
+
+    def clear(self) -> None:
+        self.__init__()
+
+    def pop(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            value = self[key]
+        except KeyError as e:
+            if default:
+                return default
+            raise e
+        self.__delitem__(key)
+        return value
+
+    def update(self, other: Dictionary | Iterable = None, **kwargs) -> None:
+        if not other:
+            for key, value in kwargs.items():
+                self.__setitem__(key, value)
+        elif isinstance(other, Dictionary):
+            for node in other.hash_table:
+                if node is not None:
+                    self.__setitem__(node.key, node.value)
+        elif isinstance(other, dict):
+            for key, value in other.items():
+                self.__setitem__(key, value)
+        elif isinstance(other, Iterable):
```

Comment:
so we can get an iterable of key-value pairs, then we can simply run it in a for-loop and unpack them, no need to wrap it in try except - let the error happen if wrong data has been passed:
```
            for key, value in other:
                self[key] = value

Reply:
I changed it but what if we have wrong Iterable object. Or it`s ok if python raise 'ValueError: not enough values to unpack (expected 2, got 1)'?

Reply:
Yeah in this case is ok, python will not allow unpacking operation or will not be able to iterate over wrong data type.
You could add custom exception of course but this method has already too much going on